atlasdb-commons/src/test/java/com/palantir/common/concurrent/InterruptibleFutureTest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.common.concurrent;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.internal.matchers.ThrowableCauseMatcher.hasCause;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RunnableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Tests for {@link InterruptibleFuture}.
+ */
+public class InterruptibleFutureTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private ExecutorService executor;
+
+    @Before
+    public void before() {
+        executor = PTExecutors.newCachedThreadPool();
+    }
+
+    @After
+    public void after() throws InterruptedException {
+        executor.shutdownNow();
+        assertThat(executor.awaitTermination(1, TimeUnit.MINUTES), is(true));
+    }
+
+    @Test
+    public void testSimple() throws Exception {
+        RunnableFuture<Integer> interruptible = getInterruptible();
+        executor.execute(interruptible);
+        assertThat(interruptible.get(), is(1));
+    }
+
+    @Test
+    public void testNoStart() throws Exception {
+        RunnableFuture<Integer> interruptible = getInterruptible();
+        expectedException.expect(TimeoutException.class);
+        interruptible.get(10, TimeUnit.MILLISECONDS);
```

Comment:
This seems somewhat flaky. Why not use the `InterruptibleWithSleep` to make sure that we'll always time out?

Reply:
Oh, but we never start :)